frame/assets/src/weights.rs

```diff
@@ -58,100 +57,144 @@ pub trait WeightInfo {
+	fn force_clear_metadata() -> Weight;
+	fn force_asset_status() -> Weight;
+	fn approve_transfer() -> Weight;
+	fn transfer_approved() -> Weight;
+	fn cancel_approval() -> Weight;
+	fn force_cancel_approval() -> Weight;
 }
 
 /// Weights for pallet_assets using the Substrate node and recommended hardware.
 pub struct SubstrateWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	fn create() -> Weight {
-		(44_459_000 as Weight)
+		(48_518_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	fn force_create() -> Weight {
-		(21_480_000 as Weight)
+		(24_464_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
-	fn destroy(z: u32, ) -> Weight {
+	fn destroy(c: u32, s: u32, a: u32, ) -> Weight {
 		(0 as Weight)
-			// Standard Error: 2_000
-			.saturating_add((1_149_000 as Weight).saturating_mul(z as Weight))
-			.saturating_add(T::DbWeight::get().reads(2 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(z as Weight)))
-	}
-	fn force_destroy(z: u32, ) -> Weight {
-		(0 as Weight)
-			// Standard Error: 2_000
-			.saturating_add((1_146_000 as Weight).saturating_mul(z as Weight))
-			.saturating_add(T::DbWeight::get().reads(2 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(z as Weight)))
+			// Standard Error: 41_000
+			.saturating_add((24_652_000 as Weight).saturating_mul(c as Weight))
+			// Standard Error: 41_000
+			.saturating_add((30_831_000 as Weight).saturating_mul(s as Weight))
+			// Standard Error: 416_000
+			.saturating_add((3_333_000 as Weight).saturating_mul(a as Weight))
+			.saturating_add(T::DbWeight::get().reads(4 as Weight))
+			.saturating_add(T::DbWeight::get().reads((2 as Weight).saturating_mul(c as Weight)))
+			.saturating_add(T::DbWeight::get().reads((2 as Weight).saturating_mul(s as Weight)))
+			.saturating_add(T::DbWeight::get().writes(2 as Weight))
+			.saturating_add(T::DbWeight::get().writes((2 as Weight).saturating_mul(c as Weight)))
+			.saturating_add(T::DbWeight::get().writes((2 as Weight).saturating_mul(s as Weight)))
+			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(a as Weight)))
 	}
 	fn mint() -> Weight {
-		(32_995_000 as Weight)
+		(46_912_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(2 as Weight))
 			.saturating_add(T::DbWeight::get().writes(2 as Weight))
 	}
 	fn burn() -> Weight {
-		(29_245_000 as Weight)
+		(47_042_000 as Weight)
 			.saturating_add(T::DbWeight::get().reads(2 as Weight))
 			.saturating_add(T::DbWeight::get().writes(2 as Weight))
 	}
 	fn transfer() -> Weight {
-		(42_211_000 as Weight)
+		(71_531_000 as Weight)
```

Comment:
This still doesn't look good. Increase of 75% in time to execute...

Reply:
As far as I can tell, it's down to the fact that the new code always bumps reference in the system account, whereas the old code didn't need to if it was a zombie (which is the case with the old benchmarking). odd that bumping a reading a storage item and bumping a reference would cost so much.

Reply:
Introduced `transfer_keep_alive` which should have a lower weight since it will never result in the source account getting killed.